src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -198,5 +206,13 @@ private static int RunInteractive(string scriptName)
             var result = command.Execute();
             return result.ExitCode;
         }
+
+        private class RunCommandFactory : ICommandFactory
```

Comment:
I think it can get moved out later if it gets reused.  But since it is only used here, nesting the class makes more sense because you can easily see what code is getting called.